app/controllers/FileUploadController.scala

```diff
@@ -16,45 +16,47 @@
 
 package controllers
 
+import java.nio.file.{Files, Paths}
+
 import config.{FrontendAuthConnector, RasContext, RasContextImpl}
-import connectors.{FileUploadConnector, UserDetailsConnector}
+import connectors.UserDetailsConnector
 import play.Logger
+import play.api.mvc.{Action, Request}
 import play.api.{Configuration, Environment, Play}
-import play.api.mvc.Action
+import services.UploadService
 import uk.gov.hmrc.auth.core.AuthConnector
+import uk.gov.hmrc.http.HeaderCarrier
 
 import scala.concurrent.Future
 
 trait FileUploadController extends RasController with PageFlowController {
 
   implicit val context: RasContext = RasContextImpl
 
-  val fileUploadConnector: FileUploadConnector
+  val fileUploadService: UploadService
 
   def get = Action.async {
     implicit request =>
       isAuthorised.flatMap {
         case Right(_) =>
-          Future.successful(Ok(views.html.file_upload()))
+          getFileUploadUrl.flatMap { urlOption =>
+            urlOption match {
+              case Some(url) => Future.successful(Ok(views.html.file_upload(url)))
+              case _ => Future.successful(Redirect(routes.GlobalErrorController.get()))
+            }
+          }
         case Left(resp) =>
           Logger.debug("[FileUploadController][get] user Not authorised")
           resp
       }
   }
 
-  def post () = Action.async { implicit request =>
-    isAuthorised.flatMap{
-      case Right(_) =>
-        fileUploadConnector.getEnvelope().map{ envelope =>
-          envelope match {
-            case Some(envelope) =>
-              //extract envelope id here or redirect to link
-              Redirect(routes.FileUploadController.get())
-            case _ =>
-              Redirect(routes.FileUploadController.get())
-          }
-        }
-      case Left(res) => res
+  private def getFileUploadUrl()(implicit hc: HeaderCarrier): Future[Option[String]] = {
```

Comment:
I dont think getFileUploadUrl method is required if we are expecting only 201 in connector all other cases would be default exception